# Inferring Phylogenetic Relationships using the Smith-Waterman Algorithm and Hierarchical Clustering

**Authors:** Rafael Hidalgo, Anthony DeVito, Nesreen Salah, Aparna S. Varde, Robert W. Meredith  
**Affiliation:** Department of Computer Science & Department of Biology, Montclair State University, Montclair, NJ, United States  
**Contact:** (hidalogor2 | devitoa4 | salahn1 | vardea | meredithr)@montclair.edu

## Abstract
All biological species undergo change over time due to the evolutionary process. These changes can occur rapidly and unpredictably. It is critical to monitor changes and detect viral variants. Phylogenetic trees serve as methods to study evolutionary relationships. This paper analyzes phylogenetic trees with reference to viruses and conducts dynamic programming using the Smith-Waterman algorithm, followed by hierarchical clustering. This methodology constitutes an intelligent approach for data mining, paving the way for examining variations in SARS-Cov-2, which can help discover knowledge potentially useful in biomedicine.

## Keywords
- Algorithms
- Biological data
- Clustering
- Data analytics
- Knowledge discovery
- Healthcare
- Intelligent data mining
- Medicine
- Phylogeny
- SARS-Cov-2
- Virus
- Virology

## Introduction
In this study, intelligent data mining is applied to the RdRp (RNA-dependent RNA polymerase) gene, unique to RNA viruses. The approach harnesses the Smith-Waterman algorithm in dynamic programming, followed by hierarchical clustering using a sample real dataset. Comparisons are made by calculating the edit distance between each RdRp gene sequence, revealing which strains are closely related and informing the structure of the inferred phylogeny.

## Models and Methods
A phylogenetic tree depicts the relationship between biological lineages related by common descent. The study uses the Smith-Waterman algorithm, adapted for phylogeny of SARS-Cov-2, to apply a scoring scheme for optimal matching subsequence.

## Conclusions and Future Work
The Smith-Waterman algorithm, in conjunction with hierarchical clustering, is deployed to generate a phylogenetic tree. The study provides insights into the evolutionary relationships of various virus strains. Future research could aim to create an improved method for studying virus evolutions through intelligent data mining.

## Acknowledgments
A. Varde acknowledges NSF grant 2018575 and R. Meredith acknowledges NSF grants DEB-1556701 and DBI-1725932.

## References
A few key references include:
- A. Gorbalenya, C. Lauber, “Phylogeny of Viruses”, Reference Module in Biomedical Sciences, 2017.
- N. Lorusso, M. Shumskaya, Introduction to Phylogeny in Mega and Using Coronavirus SARS-Cov-2. Tech Rep. Kean University, NJ, 2020.
- P. Jackowiak, K. Kuls, L. Budzko, A. Mania, M. Figlerowicz, “Phylogeny and molecular evolution of the hepatitis C virus”, Infection, Genetics and Evolution, 2014.
- ... [and more]
